hello/hello_activity_choice.py

```diff
@@ -0,0 +1,121 @@
+import asyncio
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import List
+
+from temporalio import activity, workflow
+from temporalio.client import Client
+from temporalio.worker import Worker
+
+# Activities that will be called by the workflow
+
+
+@activity.defn
+async def order_apples(amount: int) -> str:
+    return f"Ordered {amount} Apples..."
+
+
+@activity.defn
+async def order_bananas(amount: int) -> str:
+    return f"Ordered {amount} Bananas..."
+
+
+@activity.defn
+async def order_cherries(amount: int) -> str:
+    return f"Ordered {amount} Cherries..."
+
+
+@activity.defn
+async def order_oranges(amount: int) -> str:
+    return f"Ordered {amount} Oranges..."
+
+
+@dataclass
+class ShoppingListItem:
+    fruit: str
+    amount: int
+
+
+@dataclass
+class ShoppingList:
+    items: List[ShoppingListItem]
+
+
+# Basic workflow that logs and invokes different activities based on input
+@workflow.defn
+class PurchaseFruitsWorkflow:
+    @workflow.run
+    async def run(self, list: ShoppingList) -> str:
+        # Order each thing on the list
+        ordered: List[str] = []
+        for item in list.items:
+            if item.fruit == "apple":
+                ordered.append(
+                    await workflow.execute_activity(
+                        order_apples,
+                        item.amount,
+                        start_to_close_timeout=timedelta(seconds=5),
+                    )
+                )
+            elif item.fruit == "banana":
+                ordered.append(
+                    await workflow.execute_activity(
+                        order_bananas,
+                        item.amount,
+                        start_to_close_timeout=timedelta(seconds=5),
+                    )
+                )
+            elif item.fruit == "cherry":
+                ordered.append(
+                    await workflow.execute_activity(
+                        order_cherries,
+                        item.amount,
+                        start_to_close_timeout=timedelta(seconds=5),
+                    )
+                )
+            elif item.fruit == "orange":
+                ordered.append(
+                    await workflow.execute_activity(
+                        order_oranges,
+                        item.amount,
+                        start_to_close_timeout=timedelta(seconds=5),
+                    )
+                )
+            else:
+                raise ValueError(f"Unrecognized fruit: {item.fruit}")
+        return "".join(ordered)
+
+
+async def main():
+    # Start client
+    client = await Client.connect("http://localhost:7233")
```

Comment:
Why HTTP? Is gRPC using HTTP endpoints?

Reply:
This is how core expects the connection string (as a URL).
The `http` protocol can be replaced with `smtp` and it still works which is not great.
I'd recommend changing this to be just address or at least verifying on the core side that the url is an HTTP(s) url.
Note that an `https` URL will cause core to connect using tls.

Reply:
> I'd recommend changing this to be just address or at least verifying on the core side that the url is an HTTP(s) url.

Can you clarify the recommendation? Are you saying in Python we should change to just address or in Core? If the former, are you saying I should prepend the scheme? How do I know whether it should be `http` or `https`? I can't use the presence of TLS config because there may be no config needed (unless I made users use an empty one to use TLS which is a bit ugly).

Most gRPC implementations require that you opt in to TLS separate from the address. And addresses accept custom schemes (see https://github.com/grpc/grpc/blob/master/doc/naming.md and `HostPort` at https://pkg.go.dev/go.temporal.io/sdk/internal#ClientOptions). I can change to anything we want, I just need to know what we want.

Reply:
I believe in Java we use the target format: 

https://grpc.github.io/grpc-java/javadoc/io/grpc/ManagedChannelBuilder.html#forTarget-java.lang.String-

Reply:
We do in Go as well. Unfortunately our Rust library does not: https://docs.rs/tonic/latest/tonic/transport/struct.Channel.html. There is some discussion of this at https://github.com/hyperium/tonic/issues/437 and other linked issues.

Basically Hyper requires a URI, even if you override it to be a unix socket. E.g. https://github.com/hyperium/tonic/blob/9ea03c2c44ad6bf76e4141e6fb40870b9b30ab63/examples/src/uds/client.rs#L17-L27

In the future we can offer non-http options and have core parse it. But in the meantime, if we allowed a scheme-less target now, we'd have to have some other way for them to opt in to TLS (like Go and Java require today).

Reply:
> Can you clarify the recommendation? Are you saying in Python we should change to just address or in Core? If the former, are you saying I should prepend the scheme? How do I know whether it should be http or https? I can't use the presence of TLS config because there may be no config needed (unless I made users use an empty one to use TLS which is a bit ugly).

In TS we accept `tls: true | TLSConfig`, you could do the same.

I'd be fine with either that or keeping a URI but validating that the scheme is http or https in core.

No strong preference.

On note for using the HostPort format is that it's consistent with the other SDKs.

Reply:
PR opened at https://github.com/temporalio/sdk-python/issues/79